### PR TITLE
Fix break in arm32 build

### PR DESCRIPTION
**Pull request by nategraf**
Address comment https://github.com/dotnet/coreclr/pull/14772#issuecomment-353255005

### CODE REVIEW

**Comment by nategraf** [comment]
I'm still working on getting my environment locally, but this should fix the issue. @BruceForstall Could you trigger a CI test for arm?

**Comment by nategraf** [comment]
Also, this may not be the best solution as I don't fully understand the differences between a native and cross arch build, so let me know if this could be more elgant

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by BruceForstall** [comment]
Adding others: @brianrob @mmitche

File: build.cmd [hunk 1]
```diff
@@ -404,13 +404,8 @@ if NOT DEFINED PYTHON (
...
 
-if /i "%__DoCrossArchBuild%"=="1" (
-    set __IntermediatesIncDir=%__CrossCompIntermediatesDir%\src\inc
-    set __IntermediatesEventingDir=%__CrossCompIntermediatesDir%\eventing
-) else (
-    set __IntermediatesIncDir=%__IntermediatesDir%\src\inc
-    set __IntermediatesEventingDir=%__IntermediatesDir%\eventing
-)
+set __IntermediatesIncDir=%__IntermediatesDir%\src\inc
+set __IntermediatesEventingDir=%__IntermediatesDir%\eventing
```
Comment (BruceForstall) [inline build.cmd]: What's the purpose of this change? It puts all intermediates in the same place, which seems wrong.

Comment (BruceForstall) [reply build.cmd]: ok, I see, you need to do both but were previously only doing one. Didn't see the context before.

---
In reply to: [158347058](https://github.com/dotnet/coreclr/pull/15604#discussion_r158347058) [](ancestors = 158347058)

Comment (nategraf) [reply build.cmd]: Essentially the idea is that I will write the generated source to the normal intermediate directory on every build, and additionally to the crossgen folder on cross arch builds. This is related to the question I ask earlier of "Do they need to be in both places within the same build?" Before I was doing either or

File: build.cmd [hunk 2]
```diff
@@ -422,6 +417,20 @@ echo Laying out dynamically generated EventPipe Implementation
 echo Laying out ETW event logging interface
 %PYTHON% -B -Wall %__SourceDir%\scripts\genEtwProvider.py --man %__SourceDir%\vm\ClrEtwAll.man --intermediate %__IntermediatesIncDir% --exc %__SourceDir%\vm\ClrEtwAllMeta.lst || exit /b 1
 
+if /i "%__DoCrossArchBuild%"=="1" (
+    echo Laying out dynamically generated files from cross arch build
+    set __CrossCompIntermediatesIncDir=%__CrossCompIntermediatesDir%\src\inc
+    set __CrossCompIntermediatesEventingDir=%__CrossCompIntermediatesDir%\eventing
+)
```
Comment (BruceForstall) [inline build.cmd]: You can keep these as one condition (only one 'if') if all references to `__CrossCompIntermediatesIncDir` and `__CrossCompIntermediatesEventingDir` use `!` not `%`, e.g., `!__CrossCompIntermediatesEventingDir!` not `%__CrossCompIntermediatesEventingDir%`. That's the "delay evaluation" form in CMD, otherwise everything gets expanded when the "if" is processed.

**Comment by BruceForstall** [comment]
It looks like you've added a hard requirement on Python 2.7. You should update the documentation (sometime, not for this build fix) stating that, as now it says any version of 3 is acceptable (and I remember seeing contributions to allow 3 to be used): https://github.com/dotnet/coreclr/blob/master/Documentation/building/windows-instructions.md

You might also need to justify why 3 isn't acceptable.

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by nategraf** [comment]
These scripts were never updated to work with python3.  When was it declared that python3 is acceptable? Because I am not aware of any point where these scripts have worked in python3

File: build.cmd (hunk 2 above)
Comment (nategraf) [reply build.cmd]: Thanks!

**Comment by nategraf** [comment]
Irrelvent due to https://github.com/dotnet/coreclr/pull/15609
